mmime/src/other.rs

```diff
@@ -40,8 +40,14 @@ pub(crate) unsafe fn strncasecmp(
 
     // s1 and s2 might not be null terminated.
 
-    let s1_slice = std::slice::from_raw_parts(s1 as *const u8, n);
-    let s2_slice = std::slice::from_raw_parts(s2 as *const u8, n);
+    let s1_slice = std::slice::from_raw_parts(
+        s1 as *const u8,
+        std::cmp::min(n, strnlen(s1 as *const i8, n)),
```

Comment:
Is `min` needed here? Why not just `strnlen`? It is guaranteed to to return more than `n` in any case.